test/widgets/text_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:ui';
+
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/text.dart';
+
+void main() {
+  group('weightVariableTextStyle', () {
+    Future<void> runCheck(
+      String description,
+      {
+        required TextStyle Function(BuildContext context) styleBuilder,
+        bool platformRequestsBold = false,
+        required List<FontVariation> expectedFontVariations,
+        required FontWeight expectedFontWeight,
+      }
+    ) async {
+      testWidgets(description, (WidgetTester tester) async {
+        await tester.pumpWidget(
+          MaterialApp(
+            home: MediaQuery(
+              data: MediaQueryData(boldText: platformRequestsBold),
+              child: Builder(builder: (context) => Text('', style: styleBuilder(context))))));
+
+        final TextStyle? style = tester.widget<Text>(find.byType(Text)).style;
+        check(style)
+            .isNotNull()
+            ..has((style) => style.inherit, 'inherit').isTrue()
+            ..has((style) => style.fontVariations, 'fontVariations')
```

Comment:
I think these would be cleaner with a little extension, outside the test code itself. It's only a tiny bit longer that way in total, and it keeps the noise out of this test code (making the logic easier to see) while keeping the boilerplate concentrated to be maximally repetitive and boring (which is good for spotting errors in it, compared with having it interspersed with non-repetitive other code as here.)

In particular, the `isTrue` here blends in pretty invisibly, whereas a line `..inherit.isTrue()` would be quite clear.

It'd be fine for such an extension to live in the individual test file (down at the bottom), when not expecting to reuse it. For this one, perhaps better yet would be to inaugurate a `test/flutter_checks.dart`, similar to the `stdlib_checks.dart` I added in #112.